src/Symfony/Component/DependencyInjection/Container.php

```diff
@@ -176,6 +177,14 @@ public function set($id, $service)
         if (null === $service) {
             unset($this->services[$id]);
         }
+
+        if (isset($this->privates[$id])) {
+            if (null === $service) {
+                @trigger_error(sprintf('Unsetting the "%s" private service from outside of the container is deprecated since Symfony 3.2 and won\'t be supported anymore in Symfony 4.0. An exception will be thrown instead in Symfony 4.0.', $id), E_USER_DEPRECATED);
+            } else {
+                @trigger_error(sprintf('Resetting the "%s" private service from outside of the container is deprecated since Symfony 3.2 and won\'t be supported anymore in Symfony 4.0. An exception will be thrown instead in Symfony 4.0.', $id), E_USER_DEPRECATED);
+            }
```

Comment:
I suggested that before when we deprecated calling get() before the container is compiled: https://github.com/symfony/symfony/pull/18728#diff-for-comment-62531184

So yes I would say let's deprecate the method entirely for services that are not synthetic.